foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/DatabasePlatform.java

```diff
@@ -3502,46 +3511,60 @@ public void writeAddColumnClause(Writer writer, AbstractSession session, TableDe
         field.appendDBString(writer, session, table);
     }
 
-     /**
-      * INTERNAL:
-      * Override this method if the platform supports storing JDBC connection user name during
-      * {@link #initializeConnectionData(Connection)}.
-      * @return Always returns {@code false}
-      */
-     public boolean supportsConnectionUserName() {
-         return false;
-     }
+    /**
+     * INTERNAL:
+     * Override this method if the platform supports storing JDBC connection user name during
+     * {@link #initializeConnectionData(Connection)}.
+     * @return Always returns {@code false}
+     */
+    public boolean supportsConnectionUserName() {
+        return false;
+    }
 
-     /**
-      * INTERNAL:
-      * Returns user name retrieved from JDBC connection.
-      * @throws UnsupportedOperationException on every single call until overridden.
-      */
-     public String getConnectionUserName() {
-         throw new UnsupportedOperationException("Connection user name is not supported.");
-     }
+    /**
+     * INTERNAL:
+     * Returns user name retrieved from JDBC connection.
+     * @throws UnsupportedOperationException on every single call until overridden.
+     */
+    public String getConnectionUserName() {
+        throw new UnsupportedOperationException("Connection user name is not supported.");
+    }
 
-     /**
-      * INTERNAL:
-      * Returns query to check whether given table exists.
-      * Returned query must be completely prepared so it can be just executed by calling code.
-      * @param table database table meta-data
-      * @return query to check whether given table exists
-      */
-     public DataReadQuery getTableExistsQuery(final TableDefinition table) {
-         String column = null;
-         for (FieldDefinition field : table.getFields()) {
-             if (column == null) {
-                 column = field.getName();
-             } else if (field.isPrimaryKey()) {
-                 column = field.getName();
-                 break;
-             }
-         }
-         final String sql = "SELECT " + column + " FROM " + table.getFullName() + " WHERE FALSE";
-         final DataReadQuery query = new DataReadQuery(sql);
-         query.setMaxRows(1);
-         return query;
-     }
+    // Value of shouldCheckResultTableExistsQuery must be false.
+    /**
+     * INTERNAL:
+     * Returns query to check whether given table exists.
+     * Query execution throws an exception when no such table exists.
+     * @param table database table meta-data
+     * @return query to check whether given table exists
+     */
+    protected DataReadQuery getTableExistsQuery(final TableDefinition table) {
+        final String sql = "SELECT 1 FROM " + table.getFullName();
```

Comment:
If I get it right, then this query:
* returns exactly 1 row if table exists
* throws error if it does not

Correct? If so, why this is not reflected in javadoc? Does it make sense to have extra `shouldCheckResultTableExistsQuery` option and condition in `checkTableExists` to enforce `!result.isEmpty()` check?

Reply:
OK, 2:1 for simple stupid version. Pushing in. :)